app/src/main/java/net/programmierecke/radiodroid2/players/exoplayer/ExoPlayerWrapper.java

```diff
@@ -100,6 +97,34 @@ public void onReceive(Context context, Intent intent) {
     };
 
 
+    public final class CustomLoadErrorHandlingPolicy
+            extends DefaultLoadErrorHandlingPolicy {
+        final SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context);
+
+        @Override
+        public long getRetryDelayMsFor(
+                int dataType,
+                long loadDurationMs,
+                IOException exception,
+                int errorCount) {
+            int retryDelay = sharedPrefs.getInt("settings_retry_delay", 100);;
```

Comment:
I think this field were in seconds, not ms, or am I wrong?

Reply:
Delay is in milliseconds and timeouts are in seconds. 
I think there was a bug related to this in the original retry mechanism in `IcyDataSource.read`.

Reply:
Ok